src/main/java/org/entando/kubernetes/validator/PluginDescriptorValidator.java

```diff
@@ -283,11 +301,33 @@ private PluginDescriptor validateEnvVarsOrThrow(PluginDescriptor descriptor) {
                         log.debug(error);
                         throw new InvalidBundleException(error);
                     }
+
+                    // does secret belong to the bundle?
+                    if (envVar.getSecretKeyRef() != null
+                            && !doesSecretBelongToTheBundle(descriptor.getBundleId(),
```

Comment:
@firegloves about `descriptor.getBundleId()` I understand this is just for convinience, but it's kinda confusing

Reply:
@w-caffiero-entando what do you mean? That do you prefer a different method name rather than `getBundleId()`?

Reply:
does the descriptor actually contains the BundleId?